File: BAEK_JOON/C_algorithm/백준_4673번_셀프넘버.c
#include <stdio.h>
int self_num(int num) {
	int sum = num;
	while (num != 0) {
		sum += num % 10;
		num /= 10;
	}
	return sum;
}

int main() {
	int i = 1, tmp;
	char arr[10001] = { 0, };
	for (; i < 10001; i++) {
		tmp = self_num(i);
		if (tmp < 10001)
			arr[tmp] = 1;
	}

	for (i = 1; i < 10001; i++) {
		if (!arr[i])
			printf("%d\n", i);
	}

	return 0;
}

File: BAEK_JOON/Python_algorithm/백준_4673번_셀프넘버.py
def self_num(num):
	s = num
	while num != 0:
		s += num % 10
		num //= 10
	return s

arr = [True for i in range(1,11000)]
for i in range(1,10000):
	arr[self_num(i)] = False
for i in range(1,10000):
    if arr[i] : print(i)

